src/__tests__/useTimeoutFn.test.ts

```diff
@@ -108,9 +110,23 @@ describe('useTimeoutFn', () => {
     const [spy, hook] = getHook(50);
 
     expect(spy).not.toHaveBeenCalled();
-    hook.rerender({ delay: 5 });
+    hook.rerender({ delay: 5, cb: spy });
 
     jest.advanceTimersByTime(5);
     expect(spy).toHaveBeenCalledTimes(1);
   });
+
+  it('should NOT reset timeout on function change', () => {
+    const [spy, hook] = getHook(50);
+
+    jest.advanceTimersByTime(25);
+    expect(spy).not.toHaveBeenCalled();
+
+    const spy2 = jest.fn();
+    hook.rerender({ delay: 5, cb: spy2 });
+
+    jest.advanceTimersByTime(25);
+    expect(spy).not.toHaveBeenCalled();
```

Comment:
One thing I don't fully understand: if it should not reset timout, then shouldn't this mock be called? As timers where advanced 25 + 25 so it reached its timeout, but it wasn't called.

Reply:
Actually there is a mistake but not in part you pointed =) re-rendered delay was wrong.
Anyways, about your question.

No, if it would reset timeout it should have been called at 75ms, not 50, but it's called on 50, so everything is okay.